verification/src/http_server/handlers/verification/sourcify/api.rs

```diff
@@ -0,0 +1,119 @@
+use crate::{VerificationResponse, VerificationResult};
+use actix_web::{error, error::Error};
+use futures::Future;
+use reqwest::Url;
+use std::sync::Arc;
+
+use super::types::{ApiFilesResponse, ApiRequest, ApiVerificationResponse, Files};
+
+#[async_trait::async_trait]
+pub(super) trait SourcifyApi {
+    async fn verification_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error>;
+
+    async fn source_files_request(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiFilesResponse, reqwest::Error>;
+}
+
+pub struct SourcifyApiClient {
+    host: Url,
+    request_timeout: u64,
+    verification_attempts: u64,
+}
+
+impl SourcifyApiClient {
+    pub fn new(host: Url, request_timeout: u64, verification_attempts: u64) -> Self {
+        Self {
+            host,
+            request_timeout,
+            verification_attempts,
+        }
+    }
+
+    async fn make_retrying_request<F, Fut, Response>(
```

Comment:
let's make this just a usual function, not the method of the api client and accept `attempts` as a parameter

Reply:
Also thought about it, but where to put this function?
Is `src/network.rs` ok ?

Reply:
`src/network.rs` is fine for me, but also you could just leave it here for the time being.

Reply:
hm, ok. will leave it here